server/src/main/java/com/paypal/selion/grid/FileExtractor.java

```diff
@@ -174,4 +174,71 @@ static String getFileNameFromPath(String name) {
         LOGGER.exiting(files.toString());
         return files;
     }
+
+    /**
+     * Installs MicrosoftWebDriver.msi file as administrator in SELION_HOME_DIR and also deletes unwanted file and
+     * directory created by Msiexec.
+     *
+     * @param msiFile
+     * @return The path of installed MicrosoftWebDriver
+     */
+
+    static String extractMsi(String msiFile) {
+        Process process = null;
+        String exeFilePath = null;
+        boolean isMsiExtracted = true;
+        try {
+            process = Runtime.getRuntime().exec(
+                    new String[] { "msiexec", "/a", msiFile, "/qn", "TARGETDIR=" + SeLionConstants.SELION_HOME_DIR });
+            process.waitFor();
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "Could not find file " + msiFile, e);
+            isMsiExtracted = false;
+        } catch (InterruptedException e) {
+            LOGGER.log(Level.SEVERE, "Exception waiting for msiexec to be finished", e);
+            isMsiExtracted = false;
+        } finally {
+            process.destroy();
+        }
+
+        if (isMsiExtracted) {
+            moveExeFileToSeLionHomeAndDeleteExtras();
+            exeFilePath = FileUtils.getFile(SeLionConstants.SELION_HOME_DIR + SeLionConstants.EDGE_DRIVER)
+                    .getAbsolutePath();
+        }
+
+        LOGGER.exiting(exeFilePath);
+        return exeFilePath;
+
+    }
+
+    static void moveExeFileToSeLionHomeAndDeleteExtras() {
```

Comment:
@msAbhilasha I'm going to go ahead and merge this but we might have a future problem here. This code will not work for any future `.msi` files. It assumes edgedriver is the `.msi`.